Makefile

```diff
@@ -455,8 +455,10 @@ test-cli: test-images cli
 
 .PHONY: clean
 clean:
-	# Delete pre-go 1.3 vendor
-	rm -Rf vendor
+ifneq ($(findstring release,$(GIT_BRANCH)),)
+	docker rm -f `docker ps -aq`
+	docker system prune -fa
+endif
```

Comment:
We should be careful with this. Perhaps add a confirmation message? We should avoid something like https://github.com/argoproj/argo/pull/1995